docs: query batching data consistency (#6903)
apollographql/router: docs/source/routing/performance/query-batching.mdx

---
title: Query Batching
subtitle: Receive query batches with the GraphOS Router

> description: Handle multiple GraphQL requests with GraphOS Router's query batching capabilities. Aggregate operations into single HTTP requests, preserving data consistency and reducing performance overhead.

Reviewer: I think we should get rid of "performance overhead". Batching shouldn't be used for perf optimisations, but rather for data consistency. In fact, most of the times batching operations is going to negatively affect performance, but if you're doing this for data consistency purposes, you are usually okay with the performance hit.

Author: ```suggestion
description: Handle multiple GraphQL requests with GraphOS Router's query batching capabilities. Aggregate operations into single HTTP requests to preserve data consistency.
```

Reviewer: ohhh yess!

---

<EnterpriseFeature />

Learn how query batching can help preserve data consistency of responses, and learn how to configure the GraphOS Router to receive query batches.

## About query batching 

> Modern applications often require several requests to render a single page. This is usually the result of a component-based architecture where individual micro-frontends (MFE) make requests separately to fetch data relevant to them. 

Reviewer: We should probably add an extra line here saying this is not a performance optimisation, but one of the solutions to data consistency.

Author: ```suggestion
Modern applications often require several requests to render a single page. This is usually the result of a component-based architecture where individual micro-frontends (MFE) make requests separately to fetch data relevant to them. 

Query batching makes response data consistent for MFEs making multiple requests per page. It primarily preserves data consistency rather than improve performance.
```
